R/functions.R

```diff
@@ -497,108 +568,68 @@ DEAnalysisMAST <- function(scdata, id, path, verbose = FALSE) {
     bigtable <- data.frame(cbind(log2.stat.result, Auc))
 
     de <- bigtable[bigtable$log2_fc > diff.cutoff, ]
-    if (verbose) {
-      dim(de)
-    }
+
+    if (verbose) dim(de)
+
+    data.1 <- data.used.log2[, cells.coord.list1, drop = FALSE]
+    data.2 <- data.used.log2[, cells.coord.list2, drop = FALSE]
+
+    groups <- c(
+      rep("Cluster_Other", length(cells.coord.list1)),
+      rep(i, length(cells.coord.list2))
+    )
+    groups <- as.character(groups)
+
     if (dim(de)[1] > 1) {
-      data.1 <- data.used.log2[, cells.coord.list1, drop = FALSE]
-      data.2 <- data.used.log2[, cells.coord.list2, drop = FALSE]
       genes.list <- rownames(de)
-      log2fold_change <- cbind(genes.list, de$log2_fc)
-      colnames(log2fold_change) <- c("gene.name", "log2fold_change")
+      # TODO discuss what those  next two lines are doing
+      #log2fold_change <- cbind(genes.list, de$log2_fc)
+      #colnames(log2fold_change) <- c("gene.name", "log2fold_change")
```

Comment:
i could not find out what 586 and 587 are doing. nothing?

Reply:
is `log2fold_change` maybe used at a later point?

Reply:
I could not find any usage at a later point

Reply:
ok then lets leave it commented out and we will see if it leads to issues during testing